cli/commands/upgrade.ts

```diff
@@ -27,8 +27,12 @@ class UpdateProvider extends Provider {
   override async upgrade(
     {}: UpgradeOptions,
   ): Promise<void> {
+    // Enable the unstable flag only if the current installation is unstable already
+    const unstable = Object.hasOwn(Deno, "dlopen") ? ["--unstable"] : [];
```

Comment:
Humm, this isnt ideal as once dlopen becomes stable upgrades will become unstable. I am struggling to see a better way however 🤔

Reply:
That won't matter much, no? If it becomes stable then the flag is unnecessary, but it will still work.